p2p/net/mock/mock_test.go

```diff
@@ -14,8 +14,8 @@ import (
 	protocol "github.com/ipfs/go-libp2p/p2p/protocol"
 	testutil "github.com/ipfs/go-libp2p/testutil"
 
-	context "gx/QmacZi9WygGK7Me8mH53pypyscHzU386aUZXpr28GZgUct/context"
 	detectrace "github.com/jbenet/go-detect-race"
+	context "gx/QmacZi9WygGK7Me8mH53pypyscHzU386aUZXpr28GZgUct/context"
```

Comment:
> doesn't need to encode protocol information

this breaks encapsulation and proper layering. it doesnt need another
protocol today, but it makes it way harder for us to evolve gx tomorrow --
supposing we had some other prefix (either someone else's protocol, or even
an ipfs protocol that made sense). embedding the assumption that `gx/` *always
means* `ipfs/` is a source of issues. either allow full paths, or gx-nested
full paths, but not gx-nested partial paths.




On Fri, Nov 20, 2015 at 3:39 PM Stephen Whitmore <notifications@github.com>
wrote:

> In p2p/net/mock/mock_test.go
> <https://github.com/ipfs/go-libp2p/pull/8#discussion_r45533043>:
>
> >  	detectrace "github.com/jbenet/go-detect-race"
> > +	context "gx/QmacZi9WygGK7Me8mH53pypyscHzU386aUZXpr28GZgUct/context"
>
> gx/ doesn't encode protocol information re retrieval; @jbenet
> <https://github.com/jbenet> makes a good point here. However, the import
> path here doesn't *need* to encode protocol information. @whyrusleeping
> <https://github.com/whyrusleeping>, I think we talked about something
> like a .git directory (so, .gx) that stored each package's local
> installation FS path (gx/<hash> here). And then the package.json would
> define that package's protocol path (e.g. /ipns/<hash>). My only qualm
> with adding another layer of nesting is that diminishes the user experience
> (try using command line tools in Java projects -- ha).
>
> gx/[ipfs|ipns]/: encodes the protocol underneath the gx/ prefix. Nice
> that it matches up with what's in package.json, but not necessary (as per
> above). I like the 1:1 matching of FS path and location here.
>
> I don't really like ipfs/hash/, since drops the gx substring. Having it
> might be nice for identifying what this folder means when looking at a
> project using gx.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipfs/go-libp2p/pull/8/files#r45533043>.
>